message/messaging_service.cc

```diff
@@ -1276,4 +1277,17 @@ future<> messaging_service::send_paxos_learn(msg_addr id, clock_type::time_point
         std::move(reply_to), std::move(shard), std::move(response_id), std::move(trace_info));
 }
 
+void messaging_service::register_hint_mutation(std::function<future<rpc::no_wait_type> (const rpc::client_info&, rpc::opt_time_point, frozen_mutation fm, std::vector<inet_address> forward,
+        inet_address reply_to, unsigned shard, response_id_type response_id, rpc::optional<std::optional<tracing::trace_info>> trace_info)>&& func) {
+    register_handler(this, netw::messaging_verb::HINT_MUTATION, std::move(func));
+}
+void messaging_service::unregister_hint_mutation() {
+    _rpc->unregister_handler(netw::messaging_verb::HINT_MUTATION);
+}
+future<> messaging_service::send_hint_mutation(msg_addr id, clock_type::time_point timeout, const frozen_mutation& fm, std::vector<inet_address> forward,
+        inet_address reply_to, unsigned shard, response_id_type response_id, std::optional<tracing::trace_info> trace_info) {
+    return send_message_oneway_timeout(this, timeout, messaging_verb::MUTATION, std::move(id), fm, std::move(forward),
+        std::move(reply_to), std::move(shard), std::move(response_id), std::move(trace_info));
```

Comment:
Some nitpicks (I'm surprised @gleb-cloudius missed these! :)):
`std::move(shard)` seems to be pointless - it's type is `unsigned`.
The same about std::move(response_id) which is uint64_t. Although I would keep this in case it changes in the future like `inet_address` did.

Reply:
I had to leave something to you!

Reply:
It seems that some other handlers have the same problem. I got rid of moves of primitive integral types (`unsigned`, `uint32_t` and `uint64_t` - left alone typedefs such as `response_id`).